### PR TITLE
test: ensure order of targetNames and primaryKeyInfo.sortKeyFieldNames

### PR DESCRIPTION
**Pull request by dpilch**
#### Description of changes

* Add test to ensure order of targetNames and primaryKeyInfo.sortKeyFieldNames

#### Codegen Paramaters Changed or Added

N/A

#### Issue #, if available

N/A

#### Description of how you validated changes

* Unit tests

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-codegen/blob/main/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-model-introspection-visitor.test.ts [hunk 1]
```diff
@@ -210,6 +210,49 @@ describe('Primary Key Info tests', () => {
...
+        sortKeyFieldThree: String!
+        value: String!
+        name: String! @primaryKey(sortKeyFields: ["sortKeyFieldOne", "sortKeyFieldTwo", "sortKeyFieldThree"])
+        sortKeyFieldOne: String!
+        enthusiasts: [Enthusiast] @manyToMany(relationName: "EnthusiastLikes")
+        sortKeyFieldTwo: String!
+      }
+    `;
+    const result = JSON.parse(getVisitor(schema, { respectPrimaryKeyAttributesOnConnectionField: true }).generate());
+    console.log(result);
```
Comment (svidgen) [inline packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-model-introspection-visitor.test.ts:233]: Is it normal to have console logs in this test suite?

File: packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-model-introspection-visitor.test.ts [hunk 2]
```diff
@@ -210,6 +210,49 @@ describe('Primary Key Info tests', () => {
...
+    const { models: { Like, EnthusiastLikes } } = result;
+    console.log(Like.primaryKeyInfo);
```
Comment (svidgen) [inline packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-model-introspection-visitor.test.ts:235]: Ditto.

File: packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-model-introspection-visitor.test.ts [hunk 3]
```diff
@@ -210,6 +210,49 @@ describe('Primary Key Info tests', () => {
...
+    expect(result).toMatchSnapshot();
```
Comment (svidgen) [inline packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-model-introspection-visitor.test.ts:234]: I'd be curious to hear what others think. But, FWIW, in a test like this, I might actually skip the snapshot. The invariants this test cares about are clamped on with other more meaningful assertions.

File: packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-model-introspection-visitor.test.ts (hunk 1 above)
Comment (dpilch) [reply packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-model-introspection-visitor.test.ts]: Oops! Fixed

**Review by svidgen — APPROVED** [review]

**Review by sundersc — APPROVED** [review]

**Review by phani-srikar — APPROVED** [review]

**Pull request merged**